### DISCUSSION TITLE
Is this based on libimobiledevice?

### DISCUSSION THREAD

**Discussion by MMachado05** [doronz88/pymobiledevice3#1287] [General]
I'm just kinda curious about the origins of this project. I can't tell if it provides a Python wrapper of sorts around the tools and libraries available from libimobiledevice, or if Python is somehow able to access those lower-level aspects of iDevices that are accessed in libimobiledevice via C. Does this project set out to implement features not found in libimobiledevice, or just to make that program easier to use? Thanks for any info?

**Comment by doronz88** [comment]
It's pure-python and implements a lot newer stuff then libimobiledevice

**Reply by MMachado05** [reply]
Interesting; maybe I'm just not familiar enough with "real-world Python," but I'm surprised it's possible to write code that allows full device backups in an iTunes-native way and manage internal data (contacts and such) etc. in such a high-level language! What kind of "newer stuff" does pymobiledevice do that libimobiledevice doesn't?

**Reply by doronz88** [reply]
just go over the examples
